apps/itest/test/itest/batch_transactions_2_test.exs

```diff
@@ -0,0 +1,376 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule BatchTransactions2Tests do
+  use Cabbage.Feature, async: true, file: "batch_transactions_2.feature"
+  @moduletag :transactions
+
+  require Logger
+
+  alias ChildChainAPI.Api.Transaction
+  alias ChildChainAPI.Connection, as: ChildChain
+  alias ChildChainAPI.Model.TransactionBatchSubmitBodySchema
+  alias ExPlasma.Transaction.Payment
+  alias Itest.Account
+  alias Itest.ApiModel.WatcherSecurityCriticalConfiguration
+  alias Itest.Client
+  alias Itest.Fee
+  alias Itest.Poller
+  alias Itest.Transactions.Currency
+  alias Itest.Transactions.Encoding
+
+  setup do
+    {:ok, _} = DebugEvents.start_link()
+
+    [{alice_address, alice_pkey}, {bob_address, bob_pkey}, {eve_address, eve_pkey}] = Account.take_accounts(3)
+
+    eth_fee =
+      Currency.ether()
+      |> Encoding.to_hex()
+      |> Fee.get_for_currency()
+      |> Map.get("amount")
+
+    %{
+      "fee" => eth_fee,
+      "Alice" => %{
+        address: alice_address,
+        pkey: "0x" <> alice_pkey,
+        gas: 0,
+        ethereum_balance: 0,
+        ethereum_initial_balance: 0,
+        child_chain_balance: 0,
+        utxos: [],
+        transaction_submit: nil,
+        receipt_hashes: []
+      },
+      "Bob" => %{
+        address: bob_address,
+        pkey: "0x" <> bob_pkey,
+        gas: 0,
+        ethereum_balance: 0,
+        ethereum_initial_balance: 0,
+        child_chain_balance: 0,
+        utxos: [],
+        transaction_submit: nil,
+        receipt_hashes: []
+      },
+      "Eve" => %{
+        address: eve_address,
+        pkey: "0x" <> eve_pkey,
+        gas: 0,
+        ethereum_balance: 0,
+        ethereum_initial_balance: 0,
+        child_chain_balance: 0,
+        utxos: [],
+        transaction_submit: nil,
+        receipt_hashes: []
+      }
+    }
+  end
+
+  defwhen ~r/^they deposit "(?<amount>[^"]+)" ETH to the root chain$/,
+          %{amount: amount},
+          state do
+    entity_1 = "Alice"
+    entity_2 = "Bob"
+    entity_3 = "Eve"
+
+    receipts =
+      Enum.map([entity_1, entity_2, entity_3], fn entity ->
+        %{address: address} = state[entity]
+
+        {:ok, receipt_hash} =
+          amount
+          |> Currency.to_wei()
+          |> Client.deposit(address, Itest.PlasmaFramework.vault(Currency.ether()))
+
+        receipt_hash
+      end)
+
+    # lets wait for the three deposits to be recognized
+    geth_block_every = 1
```

Comment:
is this variable really needed?
`Kernel.*(geth_block_every)`

If it is, can you please use a module attribute?

Reply:
this is a variable that identifies how frequent the blocks are, so - every second (as per geth setup).